Compiler.cmake

```diff
@@ -101,27 +74,13 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
     # use C++03 std and stdlib, which is the default used by all
     # software, including all MacPorts.
   elseif(NOT COMMON_USE_CXX03)
-    if(CMAKE_COMPILER_IS_CLANG)
-      set(COMMON_CXXSTD_FLAGS "-std=c++11")
-    elseif(NOT GCC_COMPILER_VERSION VERSION_LESS 4.3)
-      if(GCC_COMPILER_VERSION VERSION_LESS 4.7)
-        set(COMMON_CXXSTD_FLAGS "-std=c++0x")
-      else()
-        if(GCC_COMPILER_VERSION VERSION_LESS 4.8)
-          # http://stackoverflow.com/questions/4438084
-          add_definitions(-D_GLIBCXX_USE_NANOSLEEP)
-        endif()
-        set(COMMON_CXXSTD_FLAGS "-std=c++11")
-      endif()
+    set(COMMON_CXXSTD_FLAGS ${CXX_DIALECT_11})
```

Comment:
Note that this logic will have to change if we ever want to select C++14. Shouldn't we think of setting C++03 (or pre-C++11) flags if `COMMON_USE_CXX03` is set instead setting c++11 if it isn't set?